base/commands/job/common.go

```diff
@@ -60,89 +63,91 @@ func idToString(id int64) string {
 	return string(buf[:])
 }
 
-func terminateJob(ctx context.Context, ec plug.ExecContext, jobID int64, nameOrID string, terminateMode int32, text string, waitState int32) error {
+func terminateJob(ctx context.Context, ec plug.ExecContext, name string, terminateMode int32, text string, waitState int32) error {
+	nameOrID := ec.Args()[0]
 	wait := ec.Props().GetBool(flagWait)
 	ci, err := ec.ClientInternal(ctx)
 	if err != nil {
 		return err
 	}
-	_, stop, err := ec.ExecuteBlocking(ctx, func(ctx context.Context, sp clc.Spinner) (any, error) {
+	jidv, stop, err := ec.ExecuteBlocking(ctx, func(ctx context.Context, sp clc.Spinner) (any, error) {
 		sp.SetText(fmt.Sprintf("%s %s", text, nameOrID))
-		ec.Logger().Info("%s %s (%s)", text, nameOrID, idToString(jobID))
 		j := jet.New(ci, sp, ec.Logger())
-		return nil, j.TerminateJob(ctx, jobID, terminateMode)
+		jis, err := j.GetJobList(ctx)
+		if err != nil {
+			return nil, err
+		}
+		jm, err := NewJobNameToIDMap(jis)
+		if err != nil {
+			return nil, err
+		}
+		jid, ok := jm.GetIDForName(nameOrID)
+		if !ok {
+			return nil, jet.ErrInvalidJobID
+		}
+		ec.Logger().Info("%s %s (%s)", text, nameOrID, idToString(jid))
+		ji, ok := jm.GetInfoForID(jid)
+		if !ok {
+			return nil, jet.ErrInvalidJobID
+		}
+		var coord types.UUID
+		if ji.LightJob {
+			conns := ci.ConnectionManager().ActiveConnections()
+			if len(conns) == 0 {
+				return nil, errors.New("not connected")
+			}
+			coord = conns[0].MemberUUID()
+		}
+		return jid, j.TerminateJob(ctx, jid, terminateMode, coord)
 	})
 	if err != nil {
 		return err
 	}
 	stop()
+	verbose := ec.Props().GetBool(clc.PropertyVerbose)
+	err = nil
```

Comment:
We already returned the error if it is not nil, we don't need to set it to nil in here.

Reply:
Addressed at: https://github.com/hazelcast/hazelcast-commandline-client/pull/290/commits/d05e883adb10b872c1f6cd9577db28dfcb3bddce